servicemanager/servicemanager_test.go

```diff
@@ -0,0 +1,545 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package servicemanager_test
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aoscloud/aos_common/aoserrors"
+	"github.com/aoscloud/aos_common/image"
+	"github.com/opencontainers/go-digest"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aoscloud/aos_servicemanager/config"
+	"github.com/aoscloud/aos_servicemanager/servicemanager"
+)
+
+func init() {
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp: false,
+		TimestampFormat:  "2006-01-02 15:04:05.000",
+		FullTimestamp:    true,
+	})
+	log.SetLevel(log.DebugLevel)
+	log.SetOutput(os.Stdout)
+}
+
+/**********************************************************************************************************************
+ * Consts
+ *********************************************************************************************************************/
+
+const (
+	errorAddServiceID = "errorAddServiceID"
+	errorGetServicID  = "errorGetServicID"
+)
+
+/**********************************************************************************************************************
+ * Types
+ *********************************************************************************************************************/
+
+type testServiceInfoProvider struct {
+	Services []servicemanager.ServiceInfo
+}
+
+/**********************************************************************************************************************
+ * Init
+ *********************************************************************************************************************/
+
+func init() {
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp: false,
+		TimestampFormat:  "2006-01-02 15:04:05.000",
+		FullTimestamp:    true,
+	})
+	log.SetLevel(log.DebugLevel)
+	log.SetOutput(os.Stdout)
+}
+
+/**********************************************************************************************************************
+* Main
+**********************************************************************************************************************/
+
+func TestMain(m *testing.M) {
+	ret := m.Run()
+
+	os.Exit(ret)
+}
+
+/**********************************************************************************************************************
+* Tests
+**********************************************************************************************************************/
+
+func TestCreateSM(t *testing.T) {
```

Comment:
This test case is useless. You can extend it with getting service info functionality for example.

Reply:
removed